remoting/src/main/java/org/apache/rocketmq/remoting/transport/mqtt/Mqtt2RemotingCommandHandler.java

```diff
@@ -0,0 +1,106 @@
+     * can be handled by this encoder.
+     *
+     * @param ctx the {@link ChannelHandlerContext} which this {@link MessageToMessageDecoder}
+     * belongs to
+     * @param msg the message to decode to an other one
+     * @param out the {@link List} to which decoded messages should be added
+     * @throws Exception is thrown if an error occurs
+     */
+    @Override
+    protected void decode(ChannelHandlerContext ctx, MqttMessage msg, List<Object> out)
+            throws Exception {
+        if (!(msg instanceof MqttMessage)) {
+            return;
+        }
+        RemotingCommand requestCommand = null;
+        MqttFixedHeader mqttFixedHeader = msg.fixedHeader();
+        Object variableHeader = msg.variableHeader();
+
+        switch (msg.fixedHeader().messageType()) {
+            case CONNECT:
+                MqttConnectPayload payload = ((MqttConnectMessage) msg).payload();
+                MqttHeader mqttHeader = new MqttHeader();
+                mqttHeader.setMessageType(mqttFixedHeader.messageType());
+                mqttHeader.setDup(mqttFixedHeader.isDup());
+                mqttHeader.setQosLevel(mqttFixedHeader.qosLevel());
+                mqttHeader.setRetain(mqttFixedHeader.isRetain());
+                mqttHeader.setRemainingLength(mqttFixedHeader.remainingLength());
+
+                MqttConnectVariableHeader mqttConnectVariableHeader = (MqttConnectVariableHeader) variableHeader;
+                mqttHeader.setName(mqttConnectVariableHeader.name());
+                mqttHeader.setVersion(mqttConnectVariableHeader.version());
+                mqttHeader.setHasUserName(mqttConnectVariableHeader.hasUserName());
+                mqttHeader.setHasPassword(mqttConnectVariableHeader.hasPassword());
+                mqttHeader.setWillRetain(mqttConnectVariableHeader.isWillRetain());
+                mqttHeader.setWillQos(mqttConnectVariableHeader.willQos());
+                mqttHeader.setWillFlag(mqttConnectVariableHeader.isWillFlag());
+                mqttHeader.setCleanSession(mqttConnectVariableHeader.isCleanSession());
+                mqttHeader
+                        .setKeepAliveTimeSeconds(mqttConnectVariableHeader.keepAliveTimeSeconds());
+
+                requestCommand = RemotingCommand
+                        .createRequestCommand(1000, mqttHeader);
+                requestCommand.makeCustomHeaderToNet();
+
+                requestCommand.setBody(encode(payload));
+                out.add(requestCommand);
+            case CONNACK:
+            case DISCONNECT:
+            case PUBLISH:
+            case PUBACK:
+            case PUBREC:
+            case PUBREL:
+            case PUBCOMP:
+            case SUBSCRIBE:
+            case SUBACK:
+            case UNSUBSCRIBE:
+            case UNSUBACK:
+            case PINGREQ:
+            case PINGRESP:
+        }
+    }
+
+    private byte[] encode(Object obj) {
```

Comment:
The encode method can be replaced by encode method in the RemotingSerializable class.